### PR TITLE
Add support for dynamic vessel info

### PR DESCRIPTION
**Pull request by tiagojsag**
Until now, vessel details used a hardcoded strucutre. since we now have multiple layers, each with different vessel info details, we need to rely on each vessel layer header to tell us which field and what type it is.

### CODE REVIEW

**Comment by nerik** [comment]
Haven't tested the feature yet, are you using workspace_vms.json ?

Looking at the code, a few comments: 
- let's try to start enforcing the rule we set for ourselves: at least a short description and a bit of context in the PR description, a gif of the feature in action even better.
- it seems to me that we are messing way to much with the state in the VesselInfo view (all vessels array and all layers array). I've mocked a few suggested changes here https://github.com/Vizzuality/GlobalFishingWatch/pull/625/commits/13655a964a1847832ce64d74e78cd1906b338d1f
it's not functional, but should give you an idea. LMK what you think.
- is there a particular reason you changed the hue to a color in the workspace? Shouldn't we use hue by default (otherwise the actual vessel hue will fallback to 0 I think)

**Comment by tiagojsag** [comment]
1- done
2- commented in your commit, but I agree with you
3- I copied it over from somewhere else. we actually convert it to a hue. I can revert that, if needed

**Comment by nerik** [comment]
Thanks 👍 
Since the vessel's `tilesetId` is used for the layer lookup, it should be a problem, or am I missing something?

File: app/src/components/Map/VesselInfoPanel.jsx [hunk 1]
```diff
@@ -31,13 +31,10 @@ class VesselInfoPanel extends Component {
...
   render() {
-    // const vesselInfo = this.props.vessels.find(vessel => vessel.shownInInfoPanel === true);
-    // instead directly use:
-    const vesselInfo = this.props.currentlyShownVessel.vessel;
-
+    const vesselInfo = this.props.currentlyShownVessel;
     const status = this.props.infoPanelStatus;
 
-    if (status === null && vesselInfo === undefined) {
+    if (!status && !vesselInfo) {
```
Comment (nerik) [inline app/src/components/Map/VesselInfoPanel.jsx]: I say never, ever rely on truthy/falsy values in JS, but it seems I'm the only one 😝 
We should remember that :
- 0, null and undefined can mean very different things.
- explicit equality checks (=== undefined) also serve as type hints in modern JS engines (better perfs)
- explicit equality checks are just easier to read

File: app/src/containers/Map/VesselInfoPanel.js [hunk 2]
```diff
@@ -3,13 +3,20 @@ import VesselInfoPanel from 'components/Map/VesselInfoPanel';
...
 import { login } from 'actions/user';
 
-const mapStateToProps = state => ({
-  // layers: state.layers.workspaceLayers,
-  // vessels: state.vesselInfo.vessels,
-  currentlyShownVessel: state.vesselInfo.currentlyShownVessel,
-  infoPanelStatus: state.vesselInfo.infoPanelStatus,
-  userPermissions: state.user.userPermissions
-});
+const mapStateToProps = (state) => {
```
Comment (nerik) [inline app/src/containers/Map/VesselInfoPanel.js]: 👍

File: app/src/constants.js [hunk 3]
```diff
@@ -122,6 +122,10 @@ export const REPORT_STATUS = {
   sent: 'sent'
 };
 
+export const VESSEL_INFO_HIDDEN = 1;
+export const VESSEL_INFO_LOADING = 2;
+export const VESSEL_INFO_LOADED = 3;
```
Comment (nerik) [inline app/src/constants.js]: In that kind of things I tend to prefer an "enumerable" style, but probably a matter of taste:
```
export const VESSEL_INFO_STATUS = {
  HIDDEN: 1
  ...
} 
...
if (status === VESSEL_INFO_STATUS.HIDDEN) 
...

```

Comment (tiagojsag) [inline app/src/constants.js]: :+1:

**Pull request merged**